storage-s3-dynamodb/src/main/java/io/delta/storage/S3DynamoDBLogStore.java

```diff
@@ -267,22 +267,34 @@ private void tryEnsureTableExists(Configuration hadoopConf) throws IOException {
                 LOG.info("Waiting for `{}` table creation", tableName);
                 try {
                     Thread.sleep(1000);
-                } catch(InterruptedException e) {
+                } catch (InterruptedException e) {
                     throw new InterruptedIOException(e.getMessage());
                 }
             } else {
                 LOG.error("table `{}` status: {}", tableName, status);
                 break;  // TODO - raise exception?
             }
-        };
+        }
+        ;
+    }
+
+    private boolean isAWSCredentialsProviderSupportHadoopFSCredentials(Class<?> awsCredentialsProvider) {
+        return Arrays.stream(awsCredentialsProvider.getConstructors())
+                .anyMatch(constructor -> constructor.getParameterCount() == 1 &&
+                        Arrays.equals(constructor.getParameterTypes(), new Class[]{Configuration.class}));
     }
 
     private AmazonDynamoDBClient getClient() throws java.io.IOException {
         try {
-            final AWSCredentialsProvider auth =
-                (AWSCredentialsProvider) Class.forName(credentialsProviderName)
-                    .getConstructor()
-                    .newInstance();
+            Class<?> awsCredentialsProvider = Class.forName(credentialsProviderName);
+            final AWSCredentialsProvider auth;
+            if (isAWSCredentialsProviderSupportHadoopFSCredentials(awsCredentialsProvider))
```

Comment:
Does the S3 FileSystem follow the same pattern where it tries to use the `constructor(Configuration)` before trying the default constructor? Just want to make sure we are not missing more changes.

Reply:
S3 file system is much more complicated. \
Mainly because there are multiple implementations like s3a/s3n/s3. \
And also because S3a support use credentials from s3a url. \
There is an until method `createAWSCredentialProviderSet` in `S3aUtils` from `hadoo-aws`. \
Which actually do all magic for S3a, specifically. But I can't use it, because it has different signature in different  Hadoop version. And of course this part is not specified as Hadoop FS API, so it can be changed and delta will become incompatible with some hadoop version. Which will be really a problem.

Reply:
@vkorukanti, while this line is not exist in PR due to rename. The question itself is more than actual.
So I am open to suggestion on this area.
My intense was :

1.  Do not copy code and logic from hadoop-aws lib.
2. Do not rely on hadoop-aws utils which used to identify which credentials to use S3n/S3a/S3 
3. Do not rely on hadoop-aws utils which used to create list of providers inside S3a FileSystem.
The main reason of this intense -  do  not add additional  dependency on hadoop-aws,

Reply:
@vkorukanti this is two or even more piece of code. One for S3a, one For S3n and may be one for S3. |
All three encapsulated in it's own S3xFileSystem implementation. \
And what even worse, even for one FS it's (sometimes)differ from version to version of hadoop. \
I don't feel confident to copy(reimplement) this logic, especially if intense of it is `keep same logic as well tested hadoop-aws` lib. \
Mainly because of 3 points:
- No matter how many code I will copy right now or how many logic reimplement, next hadoop-aws update can change it.
-  DynamoDBLogStore choose FileSystem from `s3x` prefix in spark properties. So if any new FileSystem implementation will be added to hadoop-aws, this logic will be absent in DynamoDBLogStore.
- Adding any feature with intense `keep same logic as well tested hadoop-aws` will make implicit dependency between DynamoDbLoStore and hadoop-aws lib. So if we really want it, better make it explicit.

But my intense for this MR, was to do not replicate logic of how hadoop-aws work with auth and urls.
but just add ability to support additional implementation of AWS credentials provider.
So all logic which not exist in Provider, but exist on S3xFilesystem class level, just ignored by me. \
I am pretty new in delta.io community - please, fix me if this not what delta community plan.